### ISSUE TITLE
Problem with TextBox

### ISSUE THREAD

**Issue by rimariot** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#932]
I'm using your library in a WPF assembly consumed by a third-party WinForm-based application. Not being able to use App.xaml I built an App class derived from Application that loads at runtime the two DLLs of MaterialDesign and an AppResources.xaml file where I include the dictionaries (following this technique taken from several posts on the Internet). This works!
Unfortunately when I use the TextBox controls I get an exception on the textbox theme file.
Investigating, I noticed that the problem is related to the icons pack, for example, at the point when you define the style of the ContextMenu of the definition of the TextBox style. If I remove the pack icon everything works properly.
I'm wondering how I can work around this and if this is a bug or not.
Thank you.

**Comment by rimariot** [comment]
I recreated the problem by using a copy of the repository and adding a new WinForm project plus a WPF assembly DLL. The problem emerges on the following theme files:

```
MaterialDesignTheme.ListBox.xaml
MaterialDesignTheme.Menu.xaml
MaterialDesignTheme.ToolBar.xaml
```

and consequently on the **materialDesignTheme.Defaults.xaml**.

The lines in question are all referred to PickIcon when the Width and Height properties are set, for example:

```
   <ControlTemplate.Resources>
      <Style TargetType="{x: Type wpf:PackIcon}">
         <Setter Property="Width" Value="22" />
         <Setter Property="Height" Value="22" />
       </ Style>
    </ControlTemplate.Resources>
```

The VisualStudio editor reports that the **Width** and **Height** property are not recognized or are not accessible. Very strange. Commenting on these lines, everything is OK.
This seems like a problem with the **PackIcon** object.
This problem does not exist if I use the library in a WPF application project such as the Demo.

Can you help me to understand if my analysis is correct and if it is a bug?
Thank you.

**Comment by Keboo** [comment]
@rimariot can you post a repo (or a fork of the main library) that reproduces the issue? 
Can you also post the exception that you are seeing?

**Comment by MissingL-tter** [comment]
I'll try and recreate the issue and set up a repo this week.

It only occurs when building from source, not when using the nuget package.
That much I can confirm.

**Comment by Alexdara** [comment]
hello,i have find a way to fix the design time error:

    <Style x:Key="PackIconBase" TargetType="{x:Type controlzEx:PackIconBase}">
        <Setter Property="Height" Value="16" />
        <Setter Property="Width" Value="16" />
        <Setter Property="HorizontalAlignment" Value="Left" />
        <Setter Property="VerticalAlignment" Value="Top" />
        <Setter Property="IsTabStop" Value="False" />
        <Setter Property="FlowDirection" Value="LeftToRight" />
        <Setter Property="Template">
            <Setter.Value>
                <ControlTemplate TargetType="{x:Type controlzEx:PackIconBase}">
                    <Border Background="{TemplateBinding Background}" 
                            BorderBrush="{TemplateBinding BorderBrush}"
                            BorderThickness="{TemplateBinding BorderThickness}">
                        <Viewbox FlowDirection="{TemplateBinding FlowDirection}">
                            <Canvas Width="24" Height="24">
                                <Path Data="{Binding Data, RelativeSource={RelativeSource TemplatedParent}}"                                   
                                      Fill="{TemplateBinding Foreground}" />
                            </Canvas>
                        </Viewbox>
                    </Border>
                </ControlTemplate>
            </Setter.Value>
        </Setter>
    </Style>   
    <Style TargetType="{x:Type local:PackIcon}" BasedOn="{StaticResource PackIconBase}"/>

And

       <ControlTemplate.Resources>
               <Style TargetType="{x:Type controlzEx:PackIconBase}" >
                     <Setter Property="Width" Value="16" />
                      <Setter Property="Height" Value="16" />
                </Style>
         </ControlTemplate.Resources>

**Comment by Keboo** [comment]
I believe this is resolved in the latest 3.0.0 NuGet package
